packages/cdktf/test/stack.test.ts

```diff
@@ -95,13 +94,10 @@ test("stack validation returns error when provider is missing", () => {
     },
   });
 
-  const errors = Node.of(stack).validate();
+  const errors = stack.node.validate();
 
   expect(errors).toEqual([
-    expect.objectContaining({
-      message: `Found resources without a matching povider. Please make sure to add the following providers to your stack: test-provider`,
-      source: stack,
-    }),
+    `Found resources without a matching povider. Please make sure to add the following providers to your stack: test-provider`,
```

Comment:
Why did we loose the source here?

Reply:
Because it was [dropped](https://github.com/aws/constructs/blob/10.x/src/construct.ts#L477) from `constructs`.

But thinking about it, it was probably dropped because validations now are bound to a single construct, so you must have found a bug here 🤔 I might need to walk the tree to validate all childs as well and by doing so, I can re-add the source..

Reply:
Fixed the bug (now validations are run recursively while synthesizing) and re-added the source to that part of the code.
However, this has no impact on this exact thing, as this uses `Node.validate()` which does not return a source (as it is not recursively)